### PR TITLE
root: tweak build flags and wrapping flags for auxiliary programs.

### PR DESCRIPTION
**Pull request by yipengsun**
###### Motivation for this change

- The default C++ flag is set back to `c++11`. Current `c++17` breaks lots of code
    - This problem was introduced by me in the first place. Sorry!
- No longer _append_ the `PYTHONPATH`. Use `--set` instead as suggested from https://github.com/NixOS/nixpkgs/issues/22688 
- Also set `LD_LIBRARY_PATH` directly for the auxiliary programs. This is useful when inside a `root5` nix shell (which sets `LD_LIBRARY_PATH`, thus breaks the auxiliary programs)
    - This should NOT affect macOS as there `rpath` is used. But I don't have a macOS machine at hand so can't test this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/root/default.nix [hunk 1]
```diff
@@ -120,7 +120,7 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = [
     "-Drpath=ON"
-    "-DCMAKE_CXX_STANDARD=17"
+    "-DCMAKE_CXX_STANDARD=11"
```
Comment (veprbl) [inline pkgs/applications/science/misc/root/default.nix:123]: I would just remove this setting and let the compiler/build system decide what to use.

File: pkgs/applications/science/misc/root/default.nix [hunk 2]
```diff
@@ -176,7 +176,8 @@ stdenv.mkDerivation rec {
   postInstall = ''
     for prog in rootbrowse rootcp rooteventselector rootls rootmkdir rootmv rootprint rootrm rootslimtree; do
       wrapProgram "$out/bin/$prog" \
-        --prefix PYTHONPATH : "$out/lib"
+        --set PYTHONPATH "$out/lib" \
+        --set LD_LIBRARY_PATH "$out/lib"
```
Comment (veprbl) [inline pkgs/applications/science/misc/root/default.nix:180]:
```suggestion
        --set ${lib.optionalString stdenv.isDarwin "DY"}LD_LIBRARY_PATH "$out/lib"
```

Comment (yipengsun) [reply pkgs/applications/science/misc/root/default.nix:179]: I think for newer macOS the `DYLD_LIBRARY_PATH` setting is ineffective by default? But I guess there's no harm to do this.

**Comment by yipengsun** [comment]
I did a check with `nixpkgs-review` and they BUILD fine. Both `root` and `rootbrowse` runs fine:

```
❯ nix run "nixpkgs#nixpkgs-review" pr 157069
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/157069/head:refs/nixpkgs-review/1
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (15/15), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 18 (delta 7), reused 5 (delta 4), pack-reused 3
Unpacking objects: 100% (18/18), 53.40 KiB | 1.57 MiB/s, done.
From https://github.com/NixOS/nixpkgs
 * [new branch]              master                -> refs/nixpkgs-review/0
 * [new ref]                 refs/pull/157069/head -> refs/nixpkgs-review/1
$ git worktree add /home/syp/.cache/nixpkgs-review/pr-157069/nixpkgs d9e21f284317f85b3476c0043f4efea87a226c3a
Preparing worktree (detached HEAD d9e21f28431)
Updating files: 100% (29409/29409), done.
HEAD is now at d9e21f28431 smooth: 0.9.8 -> 0.9.9
$ nix-env --option system x86_64-linux -f /home/syp/.cache/nixpkgs-review/pr-157069/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff cfb94d0e276fe4e567b9e5dd7a1e15b69089bf8c
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/syp/.cache/nixpkgs-review/pr-157069/nixpkgs -qaP --xml --out-path --show-trace --meta
5 packages updated:
hepmc3 hepmc3 hepmc3 root yoda

$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/syp/.cache/nixpkgs-review/pr-157069/build.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/157069

5 packages built:
hepmc3 python310Packages.hepmc3 python39Packages.hepmc3 root yoda-with-root
```

**Comment by veprbl** [comment]
Thank you!

**Pull request merged**